lint/util.py

```diff
@@ -49,22 +49,13 @@ def get_syntax(view):
 
     or the syntax it is mapped to in the "syntax_map" setting.
     """
-    syntax_re = re.compile(r'(?i)/([^/]+)\.(?:tmLanguage|sublime-syntax)$')
-    view_syntax = view.settings().get('syntax', '')
-    mapped_syntax = ''
-
-    if view_syntax:
-        match = syntax_re.search(view_syntax)
-
-        if match:
-            view_syntax = match.group(1).lower()
-            from .persist import settings
-            mapped_syntax = settings.get(
-                'syntax_map', {}).get(view_syntax, '').lower()
-        else:
-            view_syntax = ''
-
-    return mapped_syntax or view_syntax
+    from .persist import settings
+    syntax_map = settings.get('syntax_map')
+    for syntax, selectors in syntax_map.items():
+        for s in selectors:
+            if view.match_selector(0, s):
+                return syntax
```

Comment:
I wonder whether this should return the syntax with the maximum selector match score. Like follows:

```py
try:
    syntax, score =  max(
        (max(view.match_selector(0, s) for s in selectors, default=0),
         syntax)
        for syntax, selectors in syntax_map.items()
    )
except ValueError:
    return ""
else:
    return syntax
```

Reply:
Just to be certain, we like to use `view.match_selector` not `view.score_selector()` here?

Reply:
Yes, that should be `score_selector`.

Reply:
I'll take that comprehension as a suggestion, because it is very unreadable.
Check whether you agree with my implementation of scoring based syntax selection.